src/Mapping/Driver/StaticPHPDriver.php

```diff
@@ -5,115 +5,32 @@
 namespace Doctrine\Persistence\Mapping\Driver;
 
 use Doctrine\Persistence\Mapping\ClassMetadata;
-use Doctrine\Persistence\Mapping\MappingException;
-use RecursiveDirectoryIterator;
-use RecursiveIteratorIterator;
-use ReflectionClass;
 
-use function array_unique;
-use function get_declared_classes;
-use function in_array;
-use function is_dir;
 use function method_exists;
-use function realpath;
 
 /**
  * The StaticPHPDriver calls a static loadMetadata() method on your entity
  * classes where you can manually populate the ClassMetadata instance.
  */
 class StaticPHPDriver implements MappingDriver
 {
-    /**
-     * Paths of entity directories.
-     *
-     * @var array<int, string>
-     */
-    private array $paths = [];
+    use ColocatedMappingDriver;
 
-    /**
-     * Map of all class names.
-     *
-     * @var array<int, string>
-     * @phpstan-var list<class-string>
-     */
-    private array|null $classNames = null;
-
-    /** @param array<int, string>|string $paths */
-    public function __construct(array|string $paths)
-    {
-        $this->addPaths((array) $paths);
-    }
-
-    /** @param array<int, string> $paths */
-    public function addPaths(array $paths): void
+    /** @param array<int, string>|string|ClassLocator $paths */
+    public function __construct(array|string|ClassLocator $paths)
     {
-        $this->paths = array_unique([...$this->paths, ...$paths]);
+        if ($paths instanceof ClassLocator) {
+            $this->classLocator = $paths;
+        } else {
+            $this->addPaths((array) $paths);
+        }
     }
 
     public function loadMetadataForClass(string $className, ClassMetadata $metadata): void
     {
         $className::loadMetadata($metadata);
     }
 
-    /**
-     * {@inheritDoc}
-     *
-     * @todo Same code exists in ColocatedMappingDriver, should we re-use it
```

Comment:
This deleted code is duplicate of [`FileClassLocator`](https://github.com/doctrine/persistence/blob/4.1.x/src/Mapping/Driver/FileClassLocator.php).